react/src/hooks/useHighlight.ts

```diff
@@ -0,0 +1,250 @@
+import { useThemeMode } from 'antd-style';
+import { useMemo } from 'react';
+import { codeToHtml } from 'shiki';
+import useSWR from 'swr';
+import { Md5 } from 'ts-md5';
+
+export const FALLBACK_LANG = 'txt';
+
+export const useHighlight = (text: string, lang: string) => {
```

Comment:
Shiki shows better performance in case of syntax highlighting. based code from lobe-ui. I guess that  using useSWR is the one of good solution to use codeToHtml(async)